### PR TITLE
Form gen

### PR DESCRIPTION
**Pull request by badvision**
See #90 

Introduces form generation, still have some work to do but would like to merge as it's at a good point to help get page layouts working, etc.

Test URLs:
- Before: https://main--stewart--hlxsites.hlx.live/en/about-stewart/supplier-diversity-at-stewart
- After: https://form-gen--stewart--hlxsites.hlx.live/en/about-stewart/supplier-diversity-at-stewart

### CODE REVIEW

File: blocks/form/form.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+/**
+ * loads and generated the form
+ * @param {Element} formEmbed The marker for the form embed
+ */
+export default async function decorate(block) {
+  const formId = block.textContent.trim();
+  // The form id is everything after the colon in the text
+  const formData = await fetch(`/forms/${formId}.json`);
```
Comment (shsteimer) [inline blocks/form/form.js:203]: might make sense to wrap this in an observer so we don't load form unless we need it.

Comment (badvision) [reply blocks/form/form.js]: Consider it done!

File: styles/styles.css [hunk 2]
```diff
@@ -37,6 +37,7 @@
   --clr-primary: #007dbd;
   --clr-gray: #f2f2f2;
   --clr-gray-light: #dad6d6;
+  --clr-light: #f2f2f2;
```
Comment (shsteimer) [inline styles/styles.css:40]: same as `--clr-gray` 2 lines above this.

File: scripts/scripts.js [hunk 3]
```diff
@@ -108,6 +108,17 @@ export function buildHeroBlock(main) {
   main.prepend(section);
 }
 
+function buildFormBlocks(main) {
```
Comment (shsteimer) [inline scripts/scripts.js:111]: as discussed need to revisit this, please create an issue specifc to that please

File: blocks/form/form.js [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+}
+
+/**
+ * For a franklin form post, set up the form post handler
+ * to post via AJAX and display a success or failure message
+ * @param {*} form the form element
+ * @param {*} successMessage Success message
+ * @param {*} failureMessage Failure message
+ */
+function configureFranklinFormPost(form, successMessage, failureMessage) {
```
Comment (shsteimer) [inline blocks/form/form.js:36]: need to think more about this, but im inclined to say that I don't want this in production code.

Comment (badvision) [reply blocks/form/form.js:36]: Well it's posting to the spreadsheet managed by franklin and is production code, what would you rather call it?  ConfigureDefaultFormPost?

File: styles/styles.css (hunk 2 above)
Comment (badvision) [reply styles/styles.css]: OK fixed. :D

**Pull request closed without merging**